app/code/community/Bolt/Boltpay/Helper/ApiTrait.php

```diff
@@ -71,31 +82,20 @@ private function verify_hook_api($payload, $hmacHeader)
     {
         try {
             $url = $this->getApiUrl() . "/v1/merchant/verify_signature";
-
             $key = Mage::helper('core')->decrypt(Mage::getStoreConfig('payment/boltpay/api_key'));
 
-            $ch = curl_init($url);
-
-            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-            $httpHeader = array(
-                "X-Api-Key: $key",
-                "X-Bolt-Hmac-Sha256: $hmacHeader",
-                "Content-type: application/json",
+            $httpHeader =array(
+                'X-Api-Key'=> $key,
+                'X-Bolt-Hmac-Sha256'=> $hmacHeader,
+                'X-Nonce'=> rand(100000000, 999999999)
             );
-            curl_setopt($ch, CURLOPT_HTTPHEADER, $httpHeader);
-            curl_setopt($ch, CURLOPT_POST, true);
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
+
             $this->addMetaData(array('BOLT API REQUEST' => array('verify-hook-api-header'=>$httpHeader)),true);
             $this->addMetaData(array('BOLT API REQUEST' => array('verify-hook-api-data'=>$payload)),true);
-            $result = curl_exec($ch);
-
-            $response = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-            $this->setCurlResultWithHeader($ch, $result);
-
-            $resultJSON = $this->getCurlJSONBody();
+            $resultJSON = (string)$this->getApiClient()->post($url,$payload,$httpHeader)->getBody();
             $this->addMetaData(array('BOLT API RESPONSE' => array('verify-hook-api-response'=>$resultJSON)),true);
 
-            return $response == 200;
+            return $resultJSON;
```

Comment:
Similar to before, I'm very uneasy with changing the logic here from checking for a 200 response code to checking for the existence of a response body.  We should continue to check the HTTP response code here.